apps/api/src/domain/models/photograph/commands/create-photograph/create-photograph.command-handler.ts

```diff
@@ -0,0 +1,86 @@
+import { AggregateType } from '@coscrad/api-interfaces';
+import { CommandHandler } from '@coscrad/commands';
+import { buildMultilingualTextWithSingleItem } from '../../../../../domain/common/build-multilingual-text-with-single-item';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot } from '../../../../../domain/types/ResourceType';
+import { InternalError, isInternalError } from '../../../../../lib/errors/InternalError';
+import { isNotFound } from '../../../../../lib/types/not-found';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { BaseCreateCommandHandler } from '../../../shared/command-handlers/base-create-command-handler';
+import { BaseEvent, IEventPayload } from '../../../shared/events/base-event.entity';
+import { EventRecordMetadata } from '../../../shared/events/types/EventRecordMetadata';
+import { Photograph } from '../../entities/photograph.entity';
+import { CreatePhotograph } from './create-photograph.command';
+import { PhotographCreated } from './photograph-created.event';
+
+@CommandHandler(CreatePhotograph)
+export class CreatePhotographCommandHandler extends BaseCreateCommandHandler<Photograph> {
+    protected createNewInstance({
+        aggregateCompositeIdentifier: { id },
+        title: titleText,
+        languageCodeForTitle,
+        mediaItemId,
+        photographer,
+    }: CreatePhotograph): ResultOrError<Photograph> {
+        const newInstance = new Photograph({
+            type: AggregateType.photograph,
+            id,
+            title: buildMultilingualTextWithSingleItem(titleText, languageCodeForTitle),
+            mediaItemId,
+            // TODO Deal with this properly
```

Comment:
We need to do this.

Reply:
[Here](https://www.pivotaltracker.com/story/show/186719282) is a story for this.